addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/ElmahScanRuleUnitTest.java

```diff
@@ -0,0 +1,250 @@
+        // Given
+        nano.addHandler(new ForbiddenResponse("/"));
+        HttpMessage message = getHttpMessage(URL);
+        rule.init(message, parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    public void shouldNotAlertIfOriginalFileNorElmahReturnsNon200Code() throws Exception {
+        // Given
+        String test = "/";
+        nano.addHandler(new ForbiddenResponse(test));
+        HttpMessage message = getHttpMessage(URL);
+        message.setResponseHeader("HTTP/1.1 403 Forbidden\r\n");
+        rule.init(message, parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    public void shouldAlertIfElmahFileFound() throws Exception {
+        // Given
+        nano.addHandler(new ValidResponseWithContent("/"));
+        HttpMessage message = getHttpMessage(URL);
+        rule.init(message, parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        assertThat(alertsRaised.get(0).getRisk(), is(equalTo(Alert.RISK_MEDIUM)));
+        assertThat(alertsRaised.get(0).getConfidence(), is(equalTo(Alert.CONFIDENCE_HIGH)));
+    }
+
+    @Test
+    public void shouldNotAlertIfNonElmahFileFoundStdThreshold() throws Exception {
+        // Given
+        nano.addHandler(new ValidResponseWithoutContent("/"));
+        HttpMessage message = getHttpMessage(URL);
+        rule.init(message, parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    public void shouldAlertIfNonElmahFileFoundLowThreshold() throws Exception {
+        // Given
+        nano.addHandler(new ValidResponseWithoutContent("/"));
+        HttpMessage message = getHttpMessage(URL);
+        rule.init(message, parent);
+        rule.setAlertThreshold(AlertThreshold.LOW);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        assertThat(alertsRaised.get(0).getRisk(), is(equalTo(Alert.RISK_MEDIUM)));
+        assertThat(alertsRaised.get(0).getConfidence(), is(equalTo(Alert.CONFIDENCE_LOW)));
+    }
+
+    @Test
+    public void shouldAlertIfForbiddenResponseLowThreshold() throws Exception {
+        // Given
+        nano.addHandler(new ForbiddenResponse("/"));
+        HttpMessage message = getHttpMessage(URL);
+        message.setResponseHeader("HTTP/1.1 403 Forbidden\r\n");
+        rule.init(message, parent);
+        rule.setAlertThreshold(AlertThreshold.LOW);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        assertThat(alertsRaised.get(0).getRisk(), is(equalTo(Alert.RISK_INFO)));
+        assertThat(alertsRaised.get(0).getConfidence(), is(equalTo(Alert.CONFIDENCE_LOW)));
+    }
+
+    private static class ForbiddenResponse extends NanoServerHandler {
```

Comment:
`StatusCodeResponse`? And allow to pass the status code (to test with other status codes, 401, 404..).